pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/series/builders/MaxTimeSeriesBuilder.java

```diff
@@ -39,14 +40,15 @@ public MaxTimeSeriesBuilder(String id, TimeBuckets timeBuckets, List<String> tag
 
   @Override
   public void addValueAtIndex(int timeBucketIndex, Double value) {
-    if (_values[timeBucketIndex] == null || value > _values[timeBucketIndex]) {
-      _values[timeBucketIndex] = value;
+    Double expectedValue = (value == null) ? (Double) 0.0 : value;
+    if (_values[timeBucketIndex] == null || expectedValue > _values[timeBucketIndex]) {
+        _values[timeBucketIndex] = expectedValue;
     }
   }
 
   @Override
   public void addValue(long timeValue, Double value) {
-    int timeBucketIndex = _timeBuckets.resolveIndex(timeValue);
+    int timeBucketIndex = Objects.requireNonNull(_timeBuckets).resolveIndex(timeValue);
```

Comment:
Can we avoid the requireNonNull? It will anyways lead to the same error message (NullPointerException), and it will add another function call to a hot part of the code.

Reply:
fixed.